sentry-core/src/main/java/io/sentry/core/SentryClient.java

```diff
@@ -45,8 +46,13 @@ public SentryClient(SentryOptions options, @Nullable Connection connection) {
       options.setTransport(transport);
     }
 
-    if (connection == null) {
+    ITransportGate transportGate = options.getTransportGate();
+    if (transportGate == null) {
+      transportGate = () -> true;
+      options.setTransportGate(transportGate);
```

Comment:
Can we get away with setting this? All user defined callbacks have triggered so sounds a bit late.
We can simply keep `transportGate` as a field and ignore `options.getTransportGate` within this type.

Reply:
not sure if I got it, are you saying to not have this in the `options`? the thing is, it should come from the android module because of the `Context`, so it's either in the options or parameters all the way

Reply:
I see. Makes sense. I missed the fact the value is transport from here to the Transport (if that implementation is interested in this).